pennylane/fourier/visualization.py

```diff
@@ -0,0 +1,692 @@
+            f"Coefficient array with shape {coeffs.shape} is invalid."
+        )
+
+    # Return the coefficients; we may have switched to a numpy array or added a needed extra dimension
+    return coeffs
+
+
+def _extract_data_and_labels(coeffs):
+    """Helper function for creating frequency labels and partitionining data.
+
+    Args:
+        coeffs (array[complex]): A list of sets of Fourier coefficients.
+
+    Returns:
+        (list(str), dict[str, array[complex]): The set of frequency labels, and a data
+            dictionary split into real and imaginary parts.
+    """
+    # extract the x ticks
+    nvecs = list(to_dict(coeffs[0]).keys())
+    nvecs_formatted = [format_nvec(nvec) for nvec in nvecs]
+
+    # make data
+    data = {}
+    data["real"] = np.array([[c[nvec].real for nvec in nvecs] for c in coeffs])
+    data["imag"] = np.array([[c[nvec].imag for nvec in nvecs] for c in coeffs])
+
+    return nvecs_formatted, data
+
+
+def _adjust_spine_placement(ax):
+    """Helper function to set some common axis properties when plotting."""
+    ax.xaxis.grid()
+    ax.spines["top"].set_visible(False)
+    ax.spines["bottom"].set_position("zero")
+    ax.spines["right"].set_visible(False)
+    ax.set_axisbelow(True)
+
+
+def coefficients_violin_plot(coeffs, n_inputs, ax, colour_dict=None, show_freqs=True):
+    """Plots a list of sets of Fourier coefficients as a violin plot.
+
+    Args:
+        coeffs (array[complex]): A list of sets of Fourier coefficients. The shape of the array
+            should resemble that of the output of numpy/scipy's ``fftn`` function, or
+            :func:`~.pennylane.fourier.fourier_coefficients`.
+        n_inputs (int): The number of input variables in the function.
+        ax (array[matplotlib.axes.Axes]): Axis on which to plot. Must
+            be a pair of axes from a subplot where ``sharex="row"`` and ``sharey="col"``.
+        colour_dict (dict[str, str]): A dictionary of the form {"real" : colour_string,
+            "imag" : other_colour_string} indicating which colours should be used in the plot.
+        show_freqs (bool): Whether or not to print the frequency labels on the plot axis.
+
+    Returns:
+        array[matplotlib.axes.Axes]: The axes on which the data is plotted.
+
+    **Example**
+
+    .. code-block:: python
+
+        import matplotlib as plt
+        from pennylane.fourier import fourier_coefficients, coefficients_violin_plot
+
+        f = ... # A function
+        n_inputs = ... # Number of inputs to the function
+        degree = ... # Degree to which coefficients should be calculated
+
+        # Calculate the Fourier coefficients; may be a single set or a list of
+        # multiple sets of coefficients
+        coeffs = fourier_coefficients(f, n_inputs, degree)
+
+        # Set up subplots and plot
+        fig, ax = plt.subplots(2, 1, sharey=True, figsize=(15, 4))
+        coefficients_violin_plot(coeffs, n_inputs, ax, show_freqs=True);
```

Comment:
My thoughts from the `fourier_coefficients` docstring also apply here, but with some additional suggestions:

- definitely show output 🙂 The plot looks so impressive, we should include it at the bottom here to highlight what a user can do/expect to get when using this function.

- Make sure that the codeblocks can be copied and pasted into a terminal/notebook/script and results in valid Python code. We discovered early on that we can't guarantee non-standard Python syntax will be understood by readers as placeholders, and had quite a few issues made where readers reported "Docstring example doesn't work - `SyntaxError: unknown '...'`".

Reply:
Same comment applies to docstrings below :)